### PR TITLE
Documentation for 'created-by' labels on proxies

### PR DESCRIPTION
**Pull request by elcasteel**
# Description
When we merge the gloo and gateway pods we are going to stop writing out the Proxy CRs that the gateway logic generates. On upgrades, we are going to use labels to identify which proxies to delete. This change prepends 'gloo-' to the existing created-by labels to make it clearer to users that might have copy-pasted values as part of their custom proxies that they should not copy the labels. I also added a warning in the documentation. 

# Context

We already add the label "created_by" to managed proxies but we don't do anything with it. Also the term "gateway" is very overloaded so it's not immediately obvious that "created-by: gateway" means that the content could be deleted or modified. 

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x ] I have made corresponding changes to the documentation

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/3380

### CODE REVIEW

File: docs/content/guides/dev/example-proxy-controller.md [hunk 1]
```diff
@@ -865,7 +865,9 @@ Now that we have a proxy called `my-cool-proxy`, Gloo Edge will be serving xDS c
 However, we don't actually have an Envoy instance deployed that will receive this config. In the next section, 
 we'll walk through the steps to deploy an Envoy pod wired to receive config from Gloo Edge, identifying itself as 
 `my-cool-proxy`.  
-
+### Note on CRD metadata
+Proxies that are managed by Gloo use a label named `created-by` with values that begin with `gloo`. Do not copy these 
+labels to your proxies or they may be overwritten or deleted by Gloo processes.
```
Comment (artberger) [inline docs/content/guides/dev/example-proxy-controller.md:870]: Maybe change to a warning:

```suggestion
{{% notice warning %}}
Proxies that are managed by Gloo Edge have a label named `created-by` with values that begin with `gloo`. Do not modify these labels or use the same labels for your own proxies. Gloo Edge might overwrite or delete the proxy labels. 
{{% /notice %}}
```

Comment (elcasteel) [reply docs/content/guides/dev/example-proxy-controller.md]: Updated with a slight variation on this (we're planning to use the label to delete the actual resource not just the labels)

Comment (artberger) [reply docs/content/guides/dev/example-proxy-controller.md]: ok nice, I was wondering that! Thanks

**Comment by sam-heilbron** [comment]
new flake? https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=96ec5a7a-7c65-42df-a948-9c3b14256f82

**Comment by elcasteel** [comment]
Github shows that all the checks passed on the most recent run

**Comment by elcasteel** [comment]
I opened an issue for the test flake: https://github.com/solo-io/gloo/issues/6280

**Review by artberger — APPROVED** [review]

**Pull request merged**